src/mallocTracer.cpp

```diff
@@ -108,7 +109,10 @@ void MallocTracer::patchLibraries() {
         cc->patchImport(im_malloc, (void*)malloc_hook);
         cc->patchImport(im_calloc, (void*)calloc_hook);
         cc->patchImport(im_realloc, (void*)realloc_hook);
-        cc->patchImport(im_free, (void*)free_hook);
+
+        if(!_nofree) {
+            cc->patchImport(im_free, (void*)free_hook);
```

Comment:
If we skip patching here, we won't profile `free` next time, even when `nofree` is omitted.